### PR TITLE
Fix issues with `flock` preventing to launch a daemon even if not con…

### PR DESCRIPTION
**Pull request by didier-wenzek**
Due to a combination of issues it was often required to delete the lock files used by the mappers and the agent, especially when running these daemons alternatively as `tedge` and `root`. This happens mostly by user mistakes, but leads to a bad experience and has infected installation scripts.

The root cause is that these lock files (which are not deleted when the daemon are killed) were created with no enough permissions to be used by another user.

See: https://github.com/thin-edge/thin-edge.io/issues/1726

## Proposed changes

- The lock files are now created as writable by any users.
- To acquire the lock, the file is open without the `O_CREAT` option. This is to avoid issues in case fs.protected_regular=2
- On error, the message is now more explicit on the source of the error: a true concurrent access, or an issue with missing authorization to open the file. In the former case, this is a mistake to delete the lock file!

The fix has been manually tested with and without `fs.protected_regular=1`.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

See: https://github.com/thin-edge/thin-edge.io/issues/1726

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/common/flockfile/src/unix.rs [hunk 1]
```diff
@@ -89,6 +104,15 @@ impl Flockfile {
         fs::remove_file(&self.path)?;
         Ok(())
     }
+
+    /// Create the lock file if it does not exist,
+    /// making sure any user can use this file as a lock.
+    fn create_world_writable_lock_file(path: &PathBuf) -> Result<(), io::Error> {
+        let file = File::create(path)?;
+        let permissions = fs::Permissions::from_mode(0o666);
+
```
Comment (PradeepKiruvale) [inline crates/common/flockfile/src/unix.rs:113]: Remove the empty line.

**Review by PradeepKiruvale — APPROVED** [review]
Tested, and works as expected.

File: crates/common/flockfile/src/unix.rs (hunk 1 above)
Comment (didier-wenzek) [reply crates/common/flockfile/src/unix.rs:113]: I love empty lines ;-) Seriously, I used them to separate logic blocks.

**Comment by reubenmiller** [comment]
@didier-wenzek, in the ticket description it should be `fs.protected_regular=2` not `fs.protected_regular=1`.

Do we need to worry about cleaning up the old renamed lock files now? Or is the `/run/lock/` partition always deleted on reboot?

**Comment by didier-wenzek** [comment]
> @didier-wenzek, in the ticket description it should be `fs.protected_regular=2` not `fs.protected_regular=1`.

Done

> Do we need to worry about cleaning up the old renamed lock files now? Or is the `/run/lock/` partition always deleted on reboot?

There is no guaranties for `/run/lock/` partition to be deleted on reboot, but this is generally the case as `/run` is mounted as `tmpfs`.

**Review by reubenmiller — APPROVED** [review]
Great work @didier-wenzek 🥇. Works perfectly and we didn't have to change the lock paths (so no doc changes required) 👍

**Pull request merged**